leanplum_data_export/get_messages.py

```diff
@@ -0,0 +1,80 @@
+"""Get all messages from leanplum API and save to bigquery."""
+
+import datetime
+import json
+
+import requests
+from google.cloud import bigquery
+
+LEANPLUM_API_URL = "https://api.leanplum.com/api"
+# latest version can be found at https://docs.leanplum.com/reference#get_api-action-getmessages
+LEANPLUM_API_VERSION = "1.0.6"
+
+
+class LeanplumMessageFetcher(object):
+    def __init__(self, app_id, client_key, project, bq_dataset, table_prefix, version):
+        self.app_id = app_id
+        self.client_key = client_key
+        self.project = project
+        self.bq_dataset = bq_dataset
+        self.table_prefix = table_prefix
+        self.version = version
+
+    def write_to_bq(self, date, messages):
+        bq_client = bigquery.Client(project=self.project)
+
+        table_name = f"messages_v{self.version}"
+        if self.table_prefix is not None:
+            table_name = f"{self.table_prefix}_{table_name}"
+
+        load_config = bigquery.LoadJobConfig(
+            time_partitioning=bigquery.TimePartitioning(
+                field="load_date",
+                require_partition_filter=True,
+            ),
+            create_disposition=bigquery.CreateDisposition.CREATE_IF_NEEDED,
+            write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,
+            schema_update_options=bigquery.SchemaUpdateOption.ALLOW_FIELD_ADDITION,
+        )
+        load_job = bq_client.load_table_from_json(
+            messages,
+            destination=f"{self.bq_dataset}.{table_name}${date.replace('-', '')}",
```

Comment:
Is this loading all messages (`recent = false`) into every date?

Reply:
Yes `recent` isn't well defined so I figure it's better to just load all historical messages since it's small anyway and a single day of data will contain everything.  The `load_date` column exists to keep our own history of messages in case they're modified since LP doesn't seem to do this